### PR TITLE
Add feature test for basic advocate user creation flow

### PR DESCRIPTION
**Pull request by jsugarman**
#### What
Add feature test for basic advocate user creation flow

#### Ticket

relates to [CFP-97](https://dsdmoj.atlassian.net/browse/CFP-97)

#### Why
There are bugs in this features functionality, including
a fix for one in this PR that was found using the feature test.

In addition a bug on the api-sandbox will need to extend this
feature in future PRs

### CODE REVIEW

File: features/step_definitions/manage_users_steps.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Then('I am on the manage users page') do
+  expect(@manage_users_page).to be_displayed
+end
+
+Then(/^the following user details should (not )?exist:$/) do |negate, table|
```
Comment (jsugarman) [inline features/step_definitions/manage_users_steps.rb]: remove or use `negate`

File: features/step_definitions/new_user_steps.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  fill_in label, with: text
+end
+
+And('I click govuk checkbox {string}') do |label|
+  find('.govuk-checkboxes__item label', text: label).click
+end
+
+# e.g.
+#  And I choose govuk radio "Yes" for "Get email notifications of caseworker messages on claims you created?"
+#
```
Comment (jsugarman) [inline features/step_definitions/new_user_steps.rb]: remove

File: features/step_definitions/new_user_steps.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  expect(@new_user_page).to be_displayed
+end
+
+And('I fill in {string} with {string}') do |label, text|
+  fill_in label, with: text
+end
+
+And('I click govuk checkbox {string}') do |label|
+  find('.govuk-checkboxes__item label', text: label).click
+end
```
Comment (jsugarman) [inline features/step_definitions/new_user_steps.rb]: move to common steps

File: features/step_definitions/new_user_steps.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+# e.g.
+#  And I choose govuk radio "Yes" for "Get email notifications of caseworker messages on claims you created?"
+#
+And('I choose govuk radio {string} for {string}') do |label, legend|
+  within('.govuk-fieldset', text: legend) do
+    find('.govuk-radios__item label', text: label).click
+  end
+end
```
Comment (jsugarman) [inline features/step_definitions/new_user_steps.rb]: move to common steps

File: features/users/new_advocate_user.feature [hunk 5]
```diff
@@ -0,0 +1,30 @@
+@javascript @no-seed
+Feature: Create, edit and delete AGFS external users
```
Comment (jsugarman) [inline features/users/new_advocate_user.feature]: rename feature

File: features/users/new_advocate_user.feature [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    And I click the link 'Create a new user'
+    Then I am on the new users page
+    And I should see 'Add a new user to'
+
+    And I fill in 'First name' with 'Jim'
+    And I fill in 'Last name' with 'Bob'
+    And I fill in 'Email' with 'jim.bob@example.com'
+    And I fill in 'Email confirmation' with 'jim.bob@example.com'
+    And I choose govuk radio 'Yes' for 'Get email notifications of caseworker messages on claims you created?'
+    And I save and open screenshot
```
Comment (jsugarman) [inline features/users/new_advocate_user.feature]: remove

File: features/users/new_advocate_user.feature [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    And I click govuk checkbox 'Admin'
+    And I click govuk checkbox 'Advocate'
+
+    And I fill in 'Supplier number' with 'ATA2B'
+    And I save and open screenshot
+
+    When I click the button 'Save'
+    Then I am on the manage users page
+    And I should see 'User successfully created'
+    And I save and open screenshot
```
Comment (jsugarman) [inline features/users/new_advocate_user.feature]: remoe

File: features/users/new_advocate_user.feature [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    And I fill in 'Last name' with 'Bob'
+    And I fill in 'Email' with 'jim.bob@example.com'
+    And I fill in 'Email confirmation' with 'jim.bob@example.com'
+    And I choose govuk radio 'Yes' for 'Get email notifications of caseworker messages on claims you created?'
+    And I save and open screenshot
+    And I click govuk checkbox 'Admin'
+    And I click govuk checkbox 'Advocate'
+
+    And I fill in 'Supplier number' with 'ATA2B'
+    And I save and open screenshot
```
Comment (jsugarman) [inline features/users/new_advocate_user.feature]: remove

File: features/users/new_advocate_user.feature [hunk 9]
```diff
@@ -0,0 +1,26 @@
+@javascript @no-seed
```
Comment (naseberry) [inline features/users/new_advocate_user.feature:1]: Could we get some accessibility assertions in @jsugarman ?

**Review by naseberry — APPROVED** [review]
Aside from my comment, everything is seems ok 👍

**Review by jrmhaig — APPROVED** [review]
Looks OK

**Pull request merged**